bot/exts/evergreen/branding.py

```diff
@@ -0,0 +1,552 @@
+        """
+        await self.bot.wait_until_ready()
+
+        while True:
+            self.current_season = get_current_season()
+            branding_changed = await self.refresh()
+
+            if branding_changed:
+                await self.apply()
+
+            elif next(self.should_cycle):
+                await self.cycle()
+
+            until_midnight = await time_until_midnight()
+            await asyncio.sleep(until_midnight.total_seconds())
+
+    async def _info_embed(self) -> discord.Embed:
+        """Make an informative embed representing current season."""
+        info_embed = discord.Embed(description=self.current_season.description, colour=self.current_season.colour)
+
+        # If we're in a non-evergreen season, also show active months
+        if self.current_season is not SeasonBase:
+            active_months = ", ".join(m.name for m in self.current_season.months)
+            title = f"{self.current_season.season_name} ({active_months})"
+        else:
+            title = self.current_season.season_name
+
+        # Use the author field to show the season's name and avatar if available
+        info_embed.set_author(name=title, icon_url=self.avatar.download_url if self.avatar else EmptyEmbed)
+
+        banner = self.banner.path if self.banner is not None else "Unavailable"
+        info_embed.add_field(name="Banner", value=banner, inline=False)
+
+        avatar = self.avatar.path if self.avatar is not None else "Unavailable"
+        info_embed.add_field(name="Avatar", value=avatar, inline=False)
+
+        icons = await pretty_files(self.available_icons) or "Unavailable"
+        info_embed.add_field(name="Available icons", value=icons, inline=False)
+
+        # Only display cycle frequency if we're actually cycling
+        if len(self.available_icons) > 1 and Branding.cycle_frequency:
+            info_embed.set_footer(text=f"Icon cycle frequency: {Branding.cycle_frequency}")
+
+        return info_embed
+
+    async def _reset_remaining_icons(self) -> None:
+        """Set `remaining_icons` to a shuffled copy of `available_icons`."""
+        self.remaining_icons = random.sample(self.available_icons, k=len(self.available_icons))
+
+    async def _reset_should_cycle(self) -> None:
+        """
+        Reset the `should_cycle` counter based on configured frequency.
+
+        Counter will always yield False if either holds:
+            - Branding.cycle_frequency is falsey
+            - There are fewer than 2 available icons for current season
+
+        Cycling can be easily turned off, and we prevent re-uploading the same icon repeatedly.
+        """
+        if len(self.available_icons) > 1 and Branding.cycle_frequency:
+            wait_period = [False] * (Branding.cycle_frequency - 1)
+            counter = itertools.cycle(wait_period + [True])
+        else:
+            counter = itertools.cycle([False])
```

Comment:
How about using integers for the cycle doing an equality comparison against `Branding.cycle_frequency` in `_daemon_func`? This seems clearer to me than using the booleans, but it'd need other opinions.

Like

```py
# If cycle_frequency 0, it will compare against None.
# cycle_frequency will always be an int so no worries about it being None.
elif next(self.should_cycle) == Branding.cycle_frequency:
    await self.cycle()

...

if len(self.available_icons) > 1 and Branding.cycle_frequency:
    sequence = range(1, Branding.cycle_frequency + 1)
else:
    sequence = [None]

self.should_cycle = itertools.cycle(sequence) 
```

`should_cycle` wouldn't be as good of a name anymore.

Reply:
I do not see much of a benefit in using integers over booleans, as you can't really "peek" at the value anyway. The iterator construction becomes a lot cleaner in your example, but then we cannot simply check whether `should_cycle` is truthy. I see it as a bit of a trade-off, but ultimately I think your approach looks better.

Reply:
With integers, it's clearer to me where the truthiness is coming from since the comparison against the cycle frequency is visible.

Reply:
> ... it's clearer to me where the truthiness is coming from since the comparison against the cycle frequency is visible.

That's a good point. I think it's nicer from the daemon's perspective to simply be fed `True`/`False`, and not have to worry about anything else. Overall, however, using `range` leads to cleaner code. I implemented this in d7c413b. Thanks for your insight!